x-pack/plugins/lens/public/datasources/text_based/text_based_languages.tsx

```diff
@@ -105,45 +105,27 @@ export function getTextBasedDatasource({
   const getSuggestionsForVisualizeField = (
     state: TextBasedPrivateState,
     indexPatternId: string,
-    fieldName: string,
-    indexPatterns: IndexPatternMap
+    fieldName: string
   ) => {
     const context = state.initialContext;
     // on text based mode we offer suggestions for the query and not for a specific field
     if (fieldName) return [];
     if (context && 'dataViewSpec' in context && context.dataViewSpec.title && context.query) {
       const newLayerId = generateId();
-      const indexPattern = indexPatterns[indexPatternId];
-
-      const contextualFields = context.contextualFields;
-      const newColumns = contextualFields?.map((c) => {
-        let field = indexPattern?.getFieldByName(c);
-        if (!field) {
-          field = indexPattern?.fields.find((f) => f.name.includes(c));
-        }
-        const newId = generateId();
-        const type = field?.type ?? 'number';
+      const textBasedQueryColumns = context.textBasedColumns ?? [];
+      const newColumns = textBasedQueryColumns?.map((c) => {
```

Comment:
nit: should it be available by the fallback of the line above?

```suggestion
      const newColumns = textBasedQueryColumns.map((c) => {
```

Reply:
Correct!